tests/ops/models/test_privacy_preference_v2.py

```diff
@@ -0,0 +1,131 @@
+from fides.api.models.privacy_preference_v2 import (
+    ConsentIdentitiesMixin,
+    LastServedNoticeV2,
+    PrivacyPreferenceHistoryV2,
+    ServedNoticeHistoryV2,
+)
+
+
+class TestPrivacyPreferenceV2:
+    def test_privacy_notice_preference(
+        self,
+        db,
+        privacy_experience_privacy_center,
+        privacy_notice_us_ca_provide,
+        served_notice_history,
+    ):
+        preference_history_record = PrivacyPreferenceHistoryV2.create(
+            db=db,
+            data={
+                "anonymized_ip_address": "92.158.1.0",
+                "email": "test@email.com",
+                "method": "button",
+                "privacy_experience_config_history_id": privacy_experience_privacy_center.experience_config.experience_config_history_id,
+                "privacy_experience_id": privacy_experience_privacy_center.id,
+                "preference": "opt_in",
+                "privacy_notice_history_id": privacy_notice_us_ca_provide.histories[
+                    0
+                ].id,
+                "request_origin": "privacy_center",
+                "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/324.42 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/425.24",
+                "user_geography": "us_ca",
+                "url_recorded": "https://example.com/privacy_center",
+                "served_notice_history_id": served_notice_history.served_notice_history_id,
+            },
+            check_name=False,
+        )
+
+        assert preference_history_record.preference == UserConsentPreference.opt_in
+        assert (
+            preference_history_record.privacy_notice_history_id
+            == privacy_notice_us_ca_provide.histories[0].id
+        )
+        assert (
+            preference_history_record.privacy_notice_history
+            == privacy_notice_us_ca_provide.histories[0]
+        )
+
+        preference_history_record.delete(db)
+
+    def test_tcf_preference(
+        self,
+        db,
+        privacy_experience_france_overlay,
+    ):
+        preference_history_record = PrivacyPreferenceHistoryV2.create(
+            db=db,
+            data={
+                "anonymized_ip_address": "92.158.1.0",
+                "email": "test@email.com",
+                "fides_user_device": "051b219f-20e4-45df-82f7-5eb68a00889f",
+                "method": "accept",
+                "privacy_experience_config_history_id": None,
+                "privacy_experience_id": privacy_experience_france_overlay.id,
+                "preference": "tcf",
+                "request_origin": "tcf_overlay",
+                "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_2) AppleWebKit/324.42 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/425.24",
+                "user_geography": "fr_idg",
+                "url_recorded": "example.com/",
+                "served_notice_history_id": "served_notice_history_test_id",
```

Comment:
i was getting a bit tripped up in the relationships between `served_notice_history_id`, `privacy_notice_history_id`, etc. i think i've worked it all out by reading through the code and your comments in the model are helpful. i just wonder if there's any way to reword some of the concepts to make it a bit clearer? i don't have any great suggestions off the top of my head, and i don't think it's worth making things more difficult to implement - your comments are already helpful. just wanted to note that it's a bit of a mental exercise, though that may be unavoidable!

Reply:
hmm not sure what to do here. 

In short, FE sends to the BE that consent was served and now I generate a single id and return it the FE.  FE should send that id to the backend on a subsequent save of their preferences. Maybe I could change the name?